swarm/src/behaviour.rs

```diff
@@ -27,6 +27,24 @@ use std::error;
 ///
 /// This trait has been designed to be composable. Multiple implementations can be combined into
 /// one that handles all the behaviours at once.
+///
+/// # Deriving `NetworkBehaviour`
+///
+/// Crate users should implement this trait by using the the `#[derive(NetworkBehaviour)]`
+/// proc macro re-exported by `libp2p` crate. The macro generates a delegating `trait`
+/// implementation for the `struct`, which delegates method calls to all trait members. Any events
+/// generated by struct members are delegated to
+/// [`NetworkBehaviourEventProcess`](trait.NetworkBehaviourEventProcess.html) implementations which
+/// are expected to be provided by the user.
+///
+/// Optionally `#[behaviour(poll_method = "poll")]` can be added on the struct to call last a custom
```

Comment:
What do you think of the following?

```suggestion
/// Optionally one can implement a custom `poll` function, which needs to be tagged with the `#[behaviour(poll_method = "poll")]` attribute, and would be called last with no parameters.
```

Reply:
Oh looks like even my sentence split and left open as I did not figure out an easy way to describe the returned type which is loooong. I'll change this to your suggestion, thanks.